config_loader.py

```diff
@@ -0,0 +1,72 @@
+import yaml
+
+'''
+Configuration keys enumeration
+'''
+
+
+class ConfigKeys:
+    DESKS_ARRAY = "desks"
+    DESK_ID = "id"
+    DESK_NAME = "name"
+    DESK_SENSOR_PIN = "sensor_pin"
+    DEBUG_MODE = "debug_mode"
+
+
+''' 
+Singleton configuration instance
+'''
+__config = None
+
+'''
+This class is responsible for loading the configuration of the whole program, and
+should be responsible for parsing the data so that it is fairly easy to access what we need when using this class
+within other classes
+'''
+
+
+class YAMLConfigLoader:
+    __yaml_file = ""
+    __property_dictionary = {}
+
+    '''
+    Initialize Config Loader with config.yaml as default file
+    '''
+    def __init__(self, yaml_file):
+        self.__yaml_file = yaml_file
+        self.reload_config()
+
+    '''
+    Loads YAML config file. Could potentially change this to SQLite in future,
+    so that a user cant go in and just change values (if this is the case, it should still be managed by us)
+    '''
+    def reload_config(self):
+
+        # Load the file as stream
+        with open(self.__yaml_file, 'r') as stream:
+            try:
+                # Parse YAML. The yaml.load function already puts it into a dictionary, so no conversion is necessary
+                self.__property_dictionary = yaml.load(stream)
+            except yaml.YAMLError as e:
+                print(e)
```

Comment:
Personally I would fail louder/terminate program instead of a print. If the YAML file isn't loaded, nothing else will work.

Reply:
Do you agree/not agree? If you do, when will you change this? If not, why not?

Reply:
I agree! It's been changed! 

If you look at the code under this line it will raise e, causing the exception to raised. This will allow us to properly handle the exception, or, if un-handled will cause the program to exit with an error.